File: 1539-kth-missing-positive-number/1539-kth-missing-positive-number.java
class Solution {
    public int findKthPositive(int[] arr, int k) {
        int[] temp = new int[2_001];
        for(int i = 0; i < temp.length; i++) temp[i] = i;
        for(int num : arr) temp[num] = 0;
        int count = 0;
        for(int num : temp) {
            if (num != 0) count++;
            if (count == k) return num;
        }
        return 0;
    }
}

File: 1539-kth-missing-positive-number/1539-kth-missing-positive-number.kt
class Solution {
fun findKthPositive(arr: IntArray, k: Int): Int {
    var missing = 0
    var i = 0
    var j = 1
    while (missing < k) {
        if (i < arr.size && arr[i] == j) {
            i++
        } else {
            missing++
        }
        j++
    }
    return j - 1

}
}